### DISCUSSION TITLE
plausible.js `strict` mode that only sends requests if the hostname is the same as the `data-domain` attribute

### DISCUSSION THREAD

**Discussion by vincentdoerig** [plausible/analytics#512] [Anything Else]
It would be great to have a `data-strict` (or similar) attribute which only sends analytics when the hostname is the same as the `data-domain` attribute. I have recently noticed an increase in pageviews because of third-party sites coping my site including the plausible snippet. Hence, I also receive the analytics from their site which skew the correct statistics. In order to avoid a breaking change, I suggest adding following syntax on the snippet to "activate" this strict mode:

`<script async defer data-domain="[domain]" data-strict src="https://plausible.io/js/plausible.js"></script>`

And an implementation in [`plausible.js`](https://github.com/plausible/analytics/blob/master/tracker/src/plausible.js) similar to this:

```js
...
var location = window.location
var document = window.document
var hostname = location.hostname

var scriptEl = document.querySelector('[src*="' + plausibleHost +'"]')
var domain = scriptEl && scriptEl.getAttribute('data-domain')
var var isStrict = scriptEl && scriptEl.getAttribute('data-strict')
var lastPage;

function trigger(eventName, options) {
  if (isStrict !== null && domain !== hostname) return console.warn('Ignoring event on different hostname than provided (strict mode)');
  ...
}
...
```

**Comment by ukutaht** [comment]
Thanks. This sort of option makes sense, although I would implement it as an addon similar to `outboundLinks` and `hashMode`.

I think this might be a duplicate of  #183

**Reply by vincentdoerig** [reply]
Oh yea, it seems to be duplicate, sorry. Having it as a different addon would definetly also work.
